### DISCUSSION TITLE
Dag start date is cleared when an automatic task retry is triggered

### DISCUSSION THREAD

**Discussion by jmorgannht** [apache/airflow#18032] [Q&A]
### Apache Airflow version

2.1.3 (latest released)

### Operating System

Ubuntu 18.04.5 LTS

### Versions of Apache Airflow Providers

apache-airflow-providers-celery==1!1.0.1
apache-airflow-providers-databricks==1!1.0.1
apache-airflow-providers-ftp==1!1.1.0
apache-airflow-providers-http==1!1.1.1
apache-airflow-providers-imap==1!1.0.1
apache-airflow-providers-microsoft-azure==1!2.0.0
apache-airflow-providers-postgres==1!1.0.2
apache-airflow-providers-sqlite==1!1.0.2

### Deployment

Other

### What happened

When a DAG is configured for task retries, the the dag run start date is correctly set if there are no task failures/retries.  When a retry occurs the dag run start date is blank in the UI and null in the API dagRuns response.  Doesn't matter if the retry is successful or not, the start date is wiped every time.  

{
    "dag_runs": [
        {
            "conf": {},
            "dag_id": "XXXX__RPAPRNCORP_AGXEU_MO_138058",
            "dag_run_id": "scheduled__2021-08-30T16:05:51+00:00",
            "end_date": "2021-09-01T21:42:07.681597+00:00",
            "execution_date": "2021-08-30T16:05:51+00:00",
            "external_trigger": false,
            **"start_date": null,**
            "state": "failed"
        }
    ],
    "total_entries": 1
}

### What you expected to happen

A task retry should have no impact on the overall dag run start date.  

### How to reproduce

Trigger a dag where at least one task has retries configured and the retry is exercised due to task failure.  

### Anything else

Happens every time.  

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by ephraimbuddy** [comment]
I was not able to reproduce this in 2.1.3(using Sequential Executor) 
Also tested in main using LocalExecutor.

**Comment by potiuk** [comment]
Converted it to discussion - @jmorgannht can you please provide more context/information ways to reproduce it ?

**Reply by jmorgannht** [reply]
OK.  Dig some more digging.  It appears to be caused from using the /dags/{dag_id}/clearTaskInstances stable API endpoint to clear failing task instances.  We have both retries and if those fail, we will clear failing tasks instances for another go for some dags.  If the retry alone on the dag task is exercised, the start date is still intact.  If the clearTaskInstances endpoint is used with the below options the start date is immediately set to null and when the dag task starts running and all the way through dag completion, the dag run start date is never set again.  
{
    "only_failed": true,
    "dry_run": false,
    "reset_dag_runs": true
}
